src/coreclr/src/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.Windows.cs

```diff
@@ -12,6 +15,47 @@ internal static partial class GlobalizationMode
 
         internal static bool UseNls { get; } = !Invariant &&
             (GetSwitchValue("System.Globalization.UseNls", "DOTNET_SYSTEM_GLOBALIZATION_USENLS") ||
-                Interop.Globalization.LoadICU() == 0);
+                !LoadIcu());
+
+        private static bool LoadIcu()
+        {
+            if (!TryGetAppLocalIcuSwitchValue(out ReadOnlySpan<char> version, out ReadOnlySpan<char> suffix))
+            {
+                return Interop.Globalization.LoadICU() != 0;
+            }
+
+            string extension = ".dll";
+            string icuucBase = "icuuc";
+            string icuinBase = "icuin";
+            IntPtr icuucLib = IntPtr.Zero;
+            IntPtr icuinLib = IntPtr.Zero;
+            Assembly assembly = Assembly.GetExecutingAssembly();
+
+            int index = version.ToString().IndexOf('.', StringComparison.Ordinal);
```

Comment:
Why is the `ToString()` necessary?  Why not just:
```C#
int index = version.IndexOf('.');
```
?  It should have the same ordinal behavior.

Reply:
Yeah, I didn't know that the default IndexOf(char) had ordinal behavior, so when trying to use the span overload that takes a `StringComparison` I had to create a Span to pass down the `'.'`.